File: template_feed/Microsoft.DotNet.Common.ItemTemplates/content/MSTest-VisualBasic-TestClass/Test1.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting

Namespace Company.TestProject1
    <TestClass>
    Public Class Test1
#If (Fixture == AssemblyInitialize)
        <AssemblyInitialize>
        Public Shared Sub AssemblyInit(ByVal testContext As TestContext)
            ' This method is called once for the test assembly, before any tests are run.
        End Sub

#End If
#If (Fixture == AssemblyCleanup)
        <AssemblyCleanup>
        Public Shared Sub AssemblyCleanup()
            ' This method is called once for the test assembly, after all tests are run.
        End Sub

#End If
#If (Fixture == ClassInitialize)
        <ClassInitialize>
        Public Shared Sub ClassInit(ByVal testContext As TestContext)
            ' This method is called once for the test class, before any tests of the class are run.
        End Sub

#End If
#If (Fixture == ClassCleanup)
        <ClassCleanup>
        Public Shared Sub ClassCleanup()
            ' This method is called once for the test class, after all tests of the class are run.
        End Sub

#End If
#If (Fixture == TestInitialize)
        <TestInitialize>
        Public Sub TestInit()
            ' This method is called before each test method.
        End Sub

#End If
#If (Fixture == TestCleanup)
        <TestCleanup>
        Public Sub TestCleanup()
            ' This method is called after each test method.
        End Sub

#End If
        <TestMethod>
        Sub TestSub()

        End Sub
    End Class
End Namespace

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.9.0/content/Playwright-MSTest-CSharp/Test1.cs
namespace Company.TestProject1;

[TestClass]
public class Test1 : PageTest
{
#if (Fixture == AssemblyInitialize)
    [AssemblyInitialize]
    public static void AssemblyInit(TestContext context)
    {
        // This method is called once for the test assembly, before any tests are run.
    }

#endif
#if (Fixture == AssemblyCleanup)
    [AssemblyCleanup]
    public static void AssemblyCleanup()
    {
        // This method is called once for the test assembly, after all tests are run.
    }

#endif
#if (Fixture == ClassInitialize)
    [ClassInitialize]
    public static void ClassInit(TestContext context)
    {
        // This method is called once for the test class, before any tests of the class are run.
    }

#endif
#if (Fixture == ClassCleanup)
    [ClassCleanup]
    public static void ClassCleanup()
    {
        // This method is called once for the test class, after all tests of the class are run.
    }

#endif
#if (Fixture == TestInitialize)
    [TestInitialize]
    public void TestInit()
    {
        // This method is called before each test method.
    }

#endif
#if (Fixture == TestCleanup)
    [TestCleanup]
    public void TestCleanup()
    {
        // This method is called after each test method.
    }

#endif
    [TestMethod]
    public async Task HomepageHasPlaywrightInTitleAndGetStartedLinkLinkingToTheIntroPage()
    {
        await Page.GotoAsync("https://playwright.dev");

        // Expect a title "to contain" a substring.
        await Expect(Page).ToHaveTitleAsync(new Regex("Playwright"));

        // create a locator
        var getStarted = Page.Locator("text=Get Started");

        // Expect an attribute "to be strictly equal" to the value.
        await Expect(getStarted).ToHaveAttributeAsync("href", "/docs/intro");

        // Click the get started link.
        await getStarted.ClickAsync();

        // Expects the URL to contain intro.
        await Expect(Page).ToHaveURLAsync(new Regex(".*intro"));
    }
}
